src/hazelcore/semantics/UHPat.re

```diff
@@ -153,6 +156,35 @@ and make_inconsistent_operand =
     (Parenthesized(set_p), u_gen);
   };
 
+let patterns_of_type =
+    (u_gen: MetaVarGen.t, ty: HTyp.t): (list(t), MetaVarGen.t) => {
+  let (hole1, u_gen) = u_gen |> new_EmptyHole;
+  let (hole2, u_gen) = u_gen |> new_EmptyHole;
+  let (hole3, u_gen) = u_gen |> new_EmptyHole;
+  let (hole4, u_gen) = u_gen |> new_EmptyHole;
+  let pattern =
+    OpSeq.(
+      switch (ty) {
+      | HTyp.Hole
+      | Unit
+      | Arrow(_, _) => []
+      | Num => [wrap(hole1), wrap(hole2)]
```

Comment:
what's happening here? for number types, I think the canonical patterns should maybe be "0" and "_"

Reply:
Yeah, I wasn't sure about this either. I feel like matching on ints is rarely done, but the only use-case I could think of is matching on response/error codes. You would probably have a separate type for this in most applications, but with `0` and `_` I think almost everyone would use `== 0` anyway

Reply:
Oh wait, it seems we don't have something like `== 0` yet in Hazel. I'll change it to your suggestion.